cli/mobycli/scout_suggest.go

```diff
@@ -74,3 +80,25 @@ func pulledImageFromArgs(args []string) string {
 	}
 	return image
 }
+
+// hubLoggedIn checks whether the user has credentials configured
+// for Docker Hub. This can be an expensive operation, so use it
+// mindfully.
+func hubLoggedIn() bool {
+	result := make(chan bool)
+	go func() {
+		hubAuth, err := config.LoadDefaultConfigFile(nil).GetAuthConfig("index.docker.io")
```

Comment:
You should use credential listing here, i.e. `config.LoadDefaultConfigFile(nil).GetCredentialsStore("").GetAll()` and then check if the map has the index.docker.io key.

A call to the `get` "endpoint" could force a login in the future

Reply:
Some users `docker login registry-1.docker.io`; we may want to check for the 'alternate' domains :joy: